R/zchunk_LB123.Electricity_china.R

```diff
@@ -0,0 +1,262 @@
+    L123.in_EJ_R_elec_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID, fuel %in% L123.pct_province_elec_F$fuel) ->
+      L123.in_EJ_province_elec_F
+    Fuel_in_Yh = unique(L123.in_EJ_R_elec_F_Yh$fuel)
+    L123.pct_province_elec_F %>%
+      filter(fuel %in% Fuel_in_Yh & year %in% HISTORICAL_YEARS) %>%
+      left_join_error_no_match(L123.in_EJ_province_elec_F, by = c('year', 'fuel')) %>%
+      mutate(elec_F = value * pct) %>%
+      select(-value, -pct, -GCAM_region_ID) ->
+      L123.in_EJ_province_elec_F
+
+    # Electricity generation outputs by fuel and province
+    L123.out_EJ_R_elec_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID, fuel %in% L123.pct_province_elec_F$fuel) ->
+      L123.out_EJ_province_elec_F
+    Fuel_out_Yh = unique(L123.in_EJ_R_elec_F_Yh$fuel)
+    L123.pct_province_elec_F %>%
+      filter(fuel %in% Fuel_out_Yh, year %in% HISTORICAL_YEARS) %>%
+      left_join_error_no_match(L123.out_EJ_province_elec_F, by = c('year', 'fuel')) %>%
+      mutate(elec_F = value * pct) %>%
+      select(-value, -pct, -GCAM_region_ID) ->
+      L123.out_EJ_province_elec_F
+
+    # ELECTRICITY - OWNUSE
+    # Electricity own use by province
+    L126.in_EJ_R_elecownuse_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) ->
+      L123.in_EJ_CHINA_ownuse
+
+    L126.out_EJ_R_elecownuse_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) ->
+      L123.out_EJ_CHINA_ownuse
+
+    L123.in_EJ_CHINA_ownuse %>%
+      left_join_error_no_match(L123.out_EJ_CHINA_ownuse, by = c("GCAM_region_ID", "sector", "fuel", "year")) %>%
+      mutate(value = value.x - value.y) %>%
+      select(-value.x, -value.y) ->
+      L123.net_EJ_CHINA_ownuse
+
+    # Then build table with each province's share of the national ownuse. Note that this is assumed invariant over time.
+    L123.out_EJ_province_elec_F %>%
+      group_by(province, year) %>%
+      summarise(value = sum(elec_F)) %>%
+      ungroup %>%
+      group_by(year) %>%
+      mutate(value = value/sum(value)) %>%
+      ungroup  %>%
+      mutate(sector = 'electricity ownuse',
+              fuel = 'electricity') %>%
+      replace_na(list(value = 0)) ->
+      L123.net_pct_province_CHINA_ownuse_elec
+
+    # Net own use = national total by each province's share
+    L123.net_EJ_CHINA_ownuse %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) ->
+      L123.net_EJ_province_ownuse_elec
+
+    L123.net_pct_province_CHINA_ownuse_elec %>%
+      left_join_error_no_match(L123.net_EJ_province_ownuse_elec, by = c("sector", "fuel", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(-value.x, -value.y) ->
+      L123.net_EJ_province_ownuse_elec
+
+    # The input of the electricity_net_ownuse sector is equal to sum of all generation (industrial CHP + electric sector)
+    L123.out_EJ_province_elecind_F <- L123.out_EJ_province_elec_F
+    names(L123.out_EJ_province_elecind_F)[5] <- 'value'
```

Comment:
can you use `rename`?
`L123.out_EJ_province_elec_F %>%`
`rename(value = xx) %>%` 
`bind_rows(L132.out_EJ_province_indchp_F) ->`